src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -45,5 +50,289 @@ protected override void InsertItem(int index, IDeltaSetItem item)
 
             base.InsertItem(index, item);
         }
+
+        /// <summary>
+        /// Patch for DeltaSet, a collection for Delta<typeparamref name="TStructuralType"/>
+        /// </summary>
+        /// <param name="originalCollection">Original collection of the Type which needs to be updated</param>
+        /// /// <returns>DeltaSet response</returns>
+        public DeltaSet<TStructuralType> Patch(ICollection<TStructuralType> originalCollection)
+        {
+            ODataAPIHandler<TStructuralType> apiHandler = new DefaultODataAPIHandler<TStructuralType>(originalCollection);
+
+            return CopyChangedValues(apiHandler);
+        }
+
+
+        /// <summary>
+        /// Patch for DeltaSet, a collection for Delta<typeparamref name="TStructuralType"/>
+        /// </summary>     
+        /// <returns>DeltaSet response</returns>
+        public DeltaSet<TStructuralType> Patch(ODataAPIHandler<TStructuralType> apiHandlerOfT, ODataAPIHandlerFactory apiHandlerFactory)
+        {         
+            Debug.Assert(apiHandlerOfT != null);
+
+            return CopyChangedValues(apiHandlerOfT, apiHandlerFactory);
+        }
+
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes")]
+        internal DeltaSet<TStructuralType> CopyChangedValues(IODataAPIHandler apiHandler, ODataAPIHandlerFactory apiHandlerFactory = null)
+        {
+            //Here we are getting the keys and using the keys to find the original object 
```

Comment:
I'd suggest you move the comment here to the method documentation

Reply:
Fixed